fix: date picker ref type

Amour1688 opened the pull request:
<!--
First of all, thank you for your contribution! 😄

New feature please send a pull request to feature branch, and rest to master branch.
Pull requests will be merged after one of the collaborators approve.
Please makes sure that these forms are filled before submitting your pull request, thank you!
-->

[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [x] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link

<!--
1. Describe the source of requirement, like related issue link.
-->

### 💡 Background and solution

<!--
1. Describe the problem and the scenario.
2. GIF or snapshot should be provided if includes UI/interactive modification.
3. How to fix the problem, and list final API implementation and usage sample if that is a new feature.
-->

### 📝 Changelog

<!--
Describe changes from the user side, and list all potential break changes or other risks.
--->

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |           |
| 🇨🇳 Chinese |           |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

MadCcc committed:
fix: not allow setState in Popconfirm if unmounted

afc163 committed:
fix: DatePicker ref type

Dunqing committed:
Docs: fix fieldNames options typo (#33241)

* docs: tree-select fieldNames error

* docs: fix docs error

* fix: docs typo

* docs: fix fieldNames

2724635499 committed:
fix: The current change of pagination triggers onChange  (#33411)

* fix: table pagination onChange error (#33374)

* test: add  test case for this (#33374)

xianxin011 committed:
docs: Update use-in-typescript.zh-CN.md (#33434)

这里有点不对
/* src/App.ts */
应该是
/* src/App.tsx */
执行npx create-react-app my-project-name --template typescript后也只有App.tsx文件

cwjTerrace committed:
docs: fix version error (#33443)

Co-authored-by: machixian <machixian@myweimai.com>

LongHaoo committed:
fix: Dropdown button destroyPopupOnHide prop not working #33433 (#33442)

Co-authored-by: zengguhao.zgh <zengguhao.zgh@alibaba-inc.com>

afc163 committed:
docs: update changelog

zombieJ committed:
test: Update snapshot

MadCcc committed:
feat: add useMountedRef hook

MadCcc committed:
chore: revert import order change

chen-jingjie committed:
fix(Divider): with text dashed border color error (#33452)

MadCcc committed:
test: add test for react memory leak warning in Popconfirm

zombieJ committed:
fix: input not support number value (#33462)

localpath committed:
docs: Update index.en-US.md (#33464)

provide correct hash link for row selection docs to selection config anchor tag

afc163 commented on components/date-picker/__tests__/type.test.tsx:
```diff
@@ -15,15 +17,14 @@ describe('DatePicker.typescript', () => {
+        props,
+        ref: React.ForwardedRef<React.Component<PickerProps<Moment>> & CommonPickerMethods>,
+      ) => <DatePicker {...props} ref={ref} />,
+    );
```
要改的不是这个呀，是要修 antd 的代码让这个例子跑通。

afc163 replied:
合并到 https://github.com/ant-design/ant-design/pull/33431

acfasj replied:
我这边暂时也没想到更好的办法, 我觉得这样改是合理的

因为 `typeof DatePicker `  是一个交叉类型, 除了`DatePicker`还有`WeekPicker `、`QuarterPicker `...等, 但是`DatePicker `ref的类型, 显然就是只有`DatePicker`自己而已. 只是这样写好像不太方便和符合直觉

MadCcc committed:
refactor: refactor useMountedRef to useMounted

MadCcc committed:
refactor: refactor useMounted to useDestroyed

MadCcc committed:
Merge pull request #33432 from MadCcc/dev

fix: not allow setState in Popconfirm if unmounted

zombieJ committed:
docs: 4.18.1 changelog (#33476)

* docs: 4.18.1 changelog

* Update CHANGELOG.zh-CN.md

Co-authored-by: afc163 <afc163@gmail.com>

Co-authored-by: afc163 <afc163@gmail.com>

afc163 committed:
docs: improve Cascader API documentation (#33484)

close #33477

afc163 committed:
docs: improve changelog

afc163 committed:
Revert "fix: The current change of pagination triggers onChange  (#33411)" (#33489)

This reverts commit 629caccaae797af67dadbce60e9b2ff9ec3215b3.

afc163 committed:
docs: release 4.18.2 (#33491)

afc163 committed:
test: add test case for #33487 (#33492)

afc163 committed:
chore: remove whitespace

afc163 committed:
fix: Form should have default validateMessages in en locale (#33511)

seki2020 committed:
fix missing words in commets (#33514)

Lol

renovate[bot] committed:
chore(deps): update dependency eslint-plugin-unicorn to v40 (#33520)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

afc163 committed:
fix: image demo test in argo-ci (#33523)

close #33512

bison92 committed:
Update README-sp_MX.md (#33535)

The other title doesn't make sense. "Medio ambiente" is the environment, as in environmental pollution, nothing to do with computers.

afc163 committed:
fix: Select tag margin issue when using tagRender (#33539)

close #33531

xqa committed:
docs: update upload-png-only demo (#33544)

[[English Template / 英文模板](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE.md)]

### 🤔 这个变动的性质是？

- [ ] 新特性提交
- [ ] 日常 bug 修复
- [ ] 站点、文档改进
- [x] 演示代码改进
- [ ] 组件样式/交互改进
- [ ] TypeScript 定义更新
- [ ] 包体积优化
- [ ] 性能优化
- [ ] 功能增强
- [ ] 国际化改进
- [ ] 重构
- [ ] 代码风格优化
- [ ] 测试用例
- [ ] 分支合并
- [ ] 其他改动（是关于什么的改动？）

### 🔗 相关 Issue

无

### 💡 需求背景和解决方案

无

### 📝 更新日志

| 语言    | 更新描述 |
| ------- | -------- |
| 🇺🇸 英文 |          |
| 🇨🇳 中文 |          |

### ☑️ 请求合并前的自查清单

⚠️ 请自检并全部**勾选全部选项**。⚠️

- [x] 文档已补充或无须补充
- [x] 代码演示已提供或无须提供
- [x] TypeScript 定义已补充或无须补充
- [x] Changelog 已提供或无须提供

afc163 committed:
docs: fix update demo

PeachScript committed:
docs: correct react-router usage for breadcrumb (#33551)

xrkffgg committed:
ci: update verify-files-modify (#33552)

PeachScript committed:
docs: correct semicolon in upload demo (#33544) (#33553)

wuifdesign committed:
fix: PageHeader state update error (#33546)

developit committed:
fix: Avoid mutating Event properties that are supposed to be getters (#33536)

* Avoid mutating Event properties that are supposed to be getters

React's synthetic Event objects have mutable `target` and `currentTarget` properties. These are an incorrect emulation of the corresponding DOM Event properties, which are _getters_ and cannot be assigned-to. AntD's input handling attempts to assign to these getters, which fails in Preact because it uses native Event objects. The solution is to replace the simple property assignments with property definitions, which is likely smaller anyway due to the existing use of `Object.create()` to construct a prototypal facade of Event objects.

* Apply suggestions from code review

* Apply suggestions from code review

Co-authored-by: afc163 <afc163@gmail.com>

afc163 committed:
perf: upgrade rc-table (#33554)

woochanleee committed:
fix: Upload types for pass generic (#33543)

* fix: Upload types for pass generic

* fix: Upload type

* refactor: Upload type

* feat: add Upload type test case

LongHaoo committed:
fix: CardGridProps should expose Div Properties like onClick #33264 (#33563)

Co-authored-by: zengguhao.zgh <zengguhao.zgh@alibaba-inc.com>

afc163 committed:
style: remove svg reset styles (#33579)

see reason: https://github.com/necolas/normalize.css/commit/004d58b2f2e0ac3d03d075f8de46ce7c8234742f

close #33573

yoyo837 committed:
chore: classnames now provides its own type definitions, so we do not need a stub types definition (#33612)

renovate[bot] committed:
chore(deps): update dependency concurrently to v7 (#33630)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

hemengke1997 committed:
fix: Table emptynode fixed display (#33610)

csjkevin committed:
docs(TreeSelect): add one missing field (#33619)

LongHaoo committed:
fix: table tree hover error #33453 (#33638)

* fix: fix the table tree hover error #33453

* Update package.json

Co-authored-by: afc163 <afc163@gmail.com>

Co-authored-by: zengguhao.zgh <zengguhao.zgh@alibaba-inc.com>
Co-authored-by: xrkffgg <xrkffgg@vip.qq.com>
Co-authored-by: afc163 <afc163@gmail.com>

MadCcc committed:
feat: menu support home & end key (#33642)

zombieJ committed:
fix: Form.Item shaking when `help` is empty string (#33644)

* fix: empty help takes the line

* test: test case

xrkffgg committed:
docs: add changelog 4.18.3 (#33639)

* docs: add changelog 4.18.3

* docs: add

* docs: add

* docs: add user

* Update CHANGELOG.en-US.md

Co-authored-by: afc163 <afc163@gmail.com>

* Update CHANGELOG.zh-CN.md

* docs: add

Co-authored-by: afc163 <afc163@gmail.com>

chenshuai2144 committed:
docs: add ProComponents cards (#33647)

* merge

* merge

* docs: add ProComponents cards

Co-authored-by: 期贤 <qixian.cs@antgroup.com>

zombieJ committed:
docs: fix changelog page build issue

afc163 committed:
docs: improve changelog of 4.18.3 (#33668)

zombieJ committed:
refactor: opt for typography Tooltip perf (#33669)

dimbslmh committed:
fix: Warning: Encountered two children with the same key, 'seperator' (#33649)

* fix: Warning: Encountered two children with the same key, 'seperator' #33183

* eslint-disable-next-line react/no-array-index-key

Co-authored-by: dimbslmh <dimsbslmh@gmail.com>

afc163 committed:
test: improve layout demo (#33653)

* test: improve layout demo

* test: improve layout demo

* update snapshot

MadCcc committed:
fix: checkbox group support options other than string (#33678)

* fix: checkbox group support options other than string

* docs: update checkbox docs

* fix: radio group support value other than string

* fix: only support number and string

* docs: update group options type

fanerge committed:
fix: Image 图片底部空白 #30825 (#33631)

whwangms committed:
fix: remove input showCount attribute from parent component (#33686)

xrkffgg committed:
ci: simple files check actions

94dreamer committed:
ci: preview-deploy download add run_id (#33680)

afc163 committed:
fix: Form validateMessages in nested ConfigProvider (#33705)

close #33691

hydraZty committed:
fix: Input can not be hidden when prefix is set (#33706)

* fix: Input can not be hidden when prefix is set

* test: jest case for #33692

Co-authored-by: hydraZty <hydra_zty@163.com>

afc163 committed:
fix: <Icon component={HomeOutlined} /> align issue (#33709)

* fix: <Icon component={HomeOutlined} /> align issue

close #33703

* update snapshot

toSayNothing committed:
fix: Steps tail shouldn't show when type is navigation and labelPlacement is vertical (fix #33681) (#33716)

* Steps tail shouldn't show when type is navigation and labelPlacement is vertical (fix #33681)

* fix: Steps tail shouldn't show when type is navigation and labelPlacement is vertical
close #33681

* fix: Steps tail shouldn't show when type is navigation and labelPlacement is vertical
close #33681

JX-Zhuang committed:
fix: <Icon component={SyncOutlined} spin /> shake issue. (#33726)

MadCcc committed:
docs: add controlled preview demo for Image (#33727)

* docs: add controlled preview demo for Image

* test: update snapshot

* test: fix lint error

mrwd2009 committed:
fix: Invalid memoized context value in LocaleProvider (#33723)

* fix: Invalid memoized context value in LocaleProvider

* Add button type in test case.

* fix: Invalid memoized context value in Anchor

yanm1ng committed:
docs(table): upgrade react-sortable-hoc usage (#33729)

Co-authored-by: 亦逊 <eason.ym@alibaba-inc.com>

fanerge committed:
fix Input及相关组件设置 hidden 时展示问题 (#33735)

* fix: Image 图片底部空白 #30825

* feat: Input、Input.search、Input.Textarea、Input.password 设置 hidden 时 所有 prefix or suffix or showCount or allowClear or addonBefore or addonAfter 都应该隐藏

* fix: lint

* fix: test

* fix: test ui

vireakkeosokvibol committed:
fix: some incorrect translation for km_KH (#33738)

* Add km_KH locale to component

* Add test and docs

* Fix typo

* Add test snapshot

* Fix translation

* Add test

afc163 committed:
docs: remove v4 release qrcode

afc163 committed:
test: fix lint

zombieJ committed:
fix: Typography error on context ellipsis (#33725)

* chore: init measure

* chore: out of space

* refactor: Multiple render

* chore: auto cut

* feat: render split

* fix: ellipsis logic of suffix

* fix: ref missing

* fix: Tooltip missing

* test: snapshot

* chore: opt for textarea

* test: back part of ellipsis

* chore: back of ellipsis logic

* ellipsis logic

* fix: init ellipsis measure

* fix: ellipsis event

* chore: clean up

* test: Update snapshot

* fix: test

* test: Update snapshot

* chore: lazy ellipsis

* fix: check css ellipsis logic

* test: Update snapshot

* test: back of coverage

* chore: clean up

* test: ignore else

* test: clean up

zombieJ committed:
fix: TreeSelect not show hightlight with keyboard (#33755)

vagusX committed:
docs: add changelog 4.18.4 (#33748)

Co-authored-by: afc163 <afc163@gmail.com>

zombieJ committed:
docs: guide of theme dev (#33756)

afc163 committed:
chore: update Cacasder demo and remove array-tree-filter (#33754)

* chore: upgrade rc-cacasder

* update snapshot

zombieJ committed:
chore: Add DingDing release bot

zombieJ committed:
chore: conch script (#33766)

afc163 committed:
fix: Tag color in dark theme (#33751)

* docs: update dark theme doc

* fix: dark tag theme

Dunqing committed:
docs: ✏️ add note comment (#33771)

Co-authored-by: dengqing <qing.deng@goldenpig.com.cn>

MadCcc committed:
fix: update prevValue of checkbox if value changed (#33753)

zombieJ committed:
fix: Cascader auto placement & scroll (#33777)

* fix: Cascader auto placement & scroll

* chore: bump rc-tree-select

zombieJ committed:
chore: try to find safe version (#33783)

LongHaoo committed:
fix: Get ref of Card component #33730 (#33784)

* fix: Get ref of Card component #33730

* add test to the card ref

Co-authored-by: zengguhao.zgh <zengguhao.zgh@alibaba-inc.com>

boomler committed:
fix(checkbox): disabled checkbox should works well with tooltip (#33772)

Co-authored-by: wanghao <wanghao.wong@bytedance.com>

afc163 committed:
style: Cascader loading icon (#33799)

MadCcc committed:
test: fix cascader test coverage (#33802)

yoyo837 committed:
chore: i18n contributing docs improve (#33760)

* chore: i18n contributing docs improve

* update

* Update docs/react/i18n.zh-CN.md

Co-authored-by: afc163 <afc163@gmail.com>

* update

* update

* fix name

* 空格

* fix name

* update

* Update docs/react/i18n.zh-CN.md

Co-authored-by: afc163 <afc163@gmail.com>

* update

* update

* update

* 添加英文版本

* update

* the

* update

* update

* fix list

* Revert "fix list"

This reverts commit c6f3e2c699d133f74e0f4ce0424c1fd4ad43dcbf.

* fix list

* 指向源文件

Co-authored-by: afc163 <afc163@gmail.com>

mrwd2009 committed:
fix: Invalid hook dependency array in Sider and Row (#33804)

renovate[bot] committed:
chore(deps): update dependency simple-git to v3 (#33817)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

eliot587-blip committed:
docs: update sentence structure in the FAQ section (#33816)

yang committed:
docs: add Plasmic for Ant (#33758)

* Add Plasmic for Ant

* Point Plasmic link to dedicated Ant page

afc163 committed:
test: fix lint

afc163 committed:
docs(:sparkles:): release 4.18.5 (#33822)

* docs(:sparkles:): release 4.18.5

* update changelog

credred committed:
fix: upload button position flashes when it changes from hidden to displayed (#33820)

* fix: fix upload button position flashes when it changes from hidden to displayed (#33819)

* fix: typo in upload

Co-authored-by: chenqiufan <chenqiufan@shiqiao.com>

mrwd2009 committed:
fix: Clear timer of delayed loading after Button is unmounted (#33858)

* fix: Clear delay loading timer after Button is unmounted

* [CodeFactor] Apply fixes to commit 16cfb89

[ci skip] [skip ci]

* Change arrow function style in test case

Co-authored-by: codefactor-io <support@codefactor.io>

MadCcc committed:
fix: tooltip should work with loading switch (#33860)

* fix: tooltip should work with loading switch

* fix: only works for switch

* chore: code clean

tangjinzhou committed:
fix: block not work when small size (#33878)

https://github.com/vueComponent/ant-design-vue/issues/5219

MadCcc committed:
fix: ellipsis should display tooltip if rows larger than 1 (#33875)

* fix: ellipsis should display tooltip if rows larger than 1

* test: add test case

Amour1688 committed:
fix: date picker ref type
